src/librustc_codegen_llvm/attributes.rs

```diff
@@ -77,9 +77,17 @@ pub fn set_instrument_function(cx: &CodegenCx<'ll, '_>, llfn: &'ll Value) {
     if cx.sess().instrument_mcount() {
         // Similar to `clang -pg` behavior. Handled by the
         // `post-inline-ee-instrument` LLVM pass.
+
+        // The function name varies on platforms.
+        // See test/CodeGen/mcount.c in clang.
+        use std::ffi::CStr;
+        let target_mcount = format!("{}{}",
+            &cx.sess().target.target.options.target_mcount, "\0");
+        let mcount_name = CStr::from_bytes_with_nul(target_mcount.as_bytes()).unwrap();
```

Comment:
@nagisa I couldn't deal with `concat!`, so didn't use `const_cstr!`. Could you give me advice?

Reply:
I think you just need to use a `CString` here.

Reply:
Okay, I pushed the change.